swift/ITSClient/Tests/ITSMobilityTests/RegionOfInterestCoordinatorTests.swift

```diff
@@ -0,0 +1,102 @@
+/*
+ * Software Name : ITSClient
+ * SPDX-FileCopyrightText: Copyright (c) Orange SA
+ * SPDX-License-Identifier: MIT
+ *
+ * This software is distributed under the MIT license,
+ * see the "LICENSE.txt" file for more details or https://opensource.org/license/MIT/
+ *
+ * Software description: Swift ITS client.
+ */
+
+import Foundation
+import Testing
+@testable import ITSMobility
+
+struct RegionOfInterestCoordinatorTests {
+    private let namespace = "default"
+
+    @Test("Updating road alarm ROI should return 9 subscriptions and 0 unsubscription the first time")
+    func updating_road_alarm_roi_should_return_9_subscriptions_and_0_unsubscription_first_time() throws {
+        // Given
+        let regionOfInterestCoordinator = RegionOfInterestCoordinator()
+
+        // When
+        let requestUpdate = regionOfInterestCoordinator.updateRoadAlarmRegionOfInterest(
+            latitude: 43.63516355648167,
+            longitude: 1.3744570239910097,
+            zoomLevel: 22,
+            namespace: namespace)
+
+        // Then
+        let unwrapRequestUpdate = try #require(requestUpdate)
+        #expect(unwrapRequestUpdate.subscriptions.count == 9)
+        #expect(unwrapRequestUpdate.unsubscriptions.isEmpty)
+    }
+
+    @Test("Updating road alarm ROI should return 9 subscriptions and 9 unsubscriptions when moving")
+    func updating_road_alarm_roi_should_return_9_subscriptions_and_9_unsubscriptions_when_moving() throws {
+        // Given
+        let regionOfInterestCoordinator = RegionOfInterestCoordinator()
+        _ = regionOfInterestCoordinator.updateRoadAlarmRegionOfInterest(
+            latitude: 43.63516355648167,
+            longitude: 1.3744570239910097,
+            zoomLevel: 22,
+            namespace: namespace)
+
+        // When
+        let requestUpdate = regionOfInterestCoordinator.updateRoadAlarmRegionOfInterest(
```

Comment:
It seems the _request update_ is using the same coordinates as the old ones ; maybe it could be better to use another location and ensure the subscriptions are the expected ones (i.e. not subscribed to olders).